src/averageAges.js

```diff
@@ -37,7 +42,19 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  if (withChildren) {
+    const memsWidthSon = people.filter(({ name, sex }) =>
+      sex === 'f' && people
+        .some(({ mother }) => mother === name));
+
+    return Math.round(((memsWidthSon.reduce((sum, { born, died }) =>
+      sum + (died - born), 0) / memsWidthSon.length) * 100)) / 100;
+  }
+
+  const mems = people.filter(({ sex }) => sex === 'f');
```

Comment:
Let's move this `withChildren` check to filter method instead of duplicating logic